pre_commit_hooks/check_google_doc_style.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+
+"""Checks doc style against https://developers.google.com/style."""
+
+__copyright__ = """
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import argparse
+import re
+import sys
+
+_IGNORE_START = "<!-- google-doc-style-ignore -->"
+_IGNORE_STOP = "<!-- google-doc-style-resume -->"
+
+# These are from https://developers.google.com/style/word-list.
+_REPLACERS = (
+    ("blacklist", "blocklist"),
+    ("cons", "disadvantages"),
+    ("e.g.", "for example"),
+    ("file name", "filename"),
+    ("filesystem", "file system"),
+    ("flag", "option"),
+    ("flags", "options"),
+    ("for instance", "for example"),
+    ("i.e.", "that is"),
+    ("pros", "advantages"),
+    ("repo", "repository"),
+    ("repos", "repositories"),
+    ("tl;dr", "to summarize"),
+    ("via", "by way of"),
+    ("vice versa", "the other way around"),
+    ("vs.", "versus"),
+    ("whitelist", "allowlist"),
+)
+
+
+def _parse_args(argv=None):
+    """Parses command-line arguments and flags."""
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "paths",
+        metavar="PATH",
+        nargs="+",
+        help="One or more paths of files to check.",
+    )
+    return parser.parse_args(args=argv)
+
+
+def build_replacers():
+    """Builds replacement regex objects."""
+    for capitalize in (True, False):
+        for before, after in _REPLACERS:
+            if capitalize:
+                before = before.capitalize()
+                after = after.capitalize()
+            # This needs to handle abbreviations, such as `i.e.`, and so checks
+            # for the full search word. It may be useful to optimize if this
+            # becomes a performance issue.  Use negative lookbehind and
+            # lookahead to ensure we have word breaks.
+            yield (r"(?<!\w)%s(?!\w)" % re.escape(before), after)
```

Comment:
I wonder if it would be simpler to just use regular expressions above in `_REPLACERS`? It would make them a touch more verbose, but would be a bit more obvious to me what is happening and would give more power. Not a big deal either way.

Reply:
I'd use a separate bit of logic if we end up wanting it; I'm also wondering if you're overlooking the casing nuance noted above. Being able to avoid boilerplate regex content is useful.

Reply:
See above -- hadn't considered all of the complexity of the *substitution* side of the capitalization issue. Anyways, seems fine for now.